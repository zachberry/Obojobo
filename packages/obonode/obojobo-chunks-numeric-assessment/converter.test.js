jest.mock('obojobo-document-engine/src/scripts/common/index', () => ({
	Registry: {
		getItemForType: () => ({
			slateToObo: jest.fn(),
			oboToSlate: jest.fn()
		})
	}
}))

jest.mock('obojobo-document-engine/src/scripts/common/models/obo-model')

import Converter from './converter'
import {
	NUMERIC_ASSESSMENT_NODE,
	NUMERIC_CHOICE_NODE,
	NUMERIC_ANSWER_NODE,
	NUMERIC_FEEDBACK_NODE
} from './constants'

describe('NumericAssessment Converter', () => {
	test('slateToObo converts a Slate node to an OboNode with no content', () => {
		const slateNode = {
			key: 'mock_key',
			type: 'mock_type',
			data: {
				get: () => null
			},
			nodes: [
				{
					type: 'invalid_type'
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('slateToObo converts a Slate node to an OboNode', () => {
		const slateNode = {
			key: 'mock_key',
			type: 'mock_type',
			data: {
				get: () => {}
			},
			nodes: [
				{
					type: NUMERIC_CHOICE_NODE,
					nodes: [
						{
							type: NUMERIC_ANSWER_NODE,
							nodes: [],
							data: {
								get: () => ({
									requirement: 'exact'
								})
							}
						},
						{
							type: NUMERIC_FEEDBACK_NODE,
							nodes: []
						}
					]
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('slateToObo converts a Slate node to an OboNode with two correct', () => {
		const slateNode = {
			key: 'mock_key',
			type: 'mock_type',
			nodes: [
				{
					type: NUMERIC_ASSESSMENT_NODE,
					nodes: [
						{
							type: NUMERIC_CHOICE_NODE,
							data: {
								get: () => {
									return { score: 100 }
								}
							}
						},
						{
							type: NUMERIC_CHOICE_NODE,
							data: {
								get: () => {
									return { score: 100 }
								}
							}
						},
						{
							type: NUMERIC_CHOICE_NODE,
							data: {
								get: () => {
									return { score: 0 }
								}
							}
						}
					]
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node with no content', () => {
		const oboNode = {
			id: 'mock_key',
			type: 'mock_type'
		}

		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mock_id',
			type: 'mock_type',
			content: {
				numericChoices: [
					{
						type: 'percent',
						score: 100,
						answer: '3',
						margin: '3',
						feedback: {
							id: 'feedback_id',
							type: NUMERIC_FEEDBACK_NODE,
							content: {},
							children: [
								{
									id: 'text_id',
									type: 'ObojoboDraft.Chunks.Text',
									content: {
										textGroup: [
											{
												data: {
													indent: 0
												},
												text: {
													value: '3222',
													styleList: []
												}
											}
										]
									},
									children: []
								}
							]
						},
						requirement: 'margin'
					},
					{
						type: 'absolute',
						score: 100,
						answer: '3',
						margin: '3',
						requirement: 'margin'
					}
				]
			}
		}

		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})
})
