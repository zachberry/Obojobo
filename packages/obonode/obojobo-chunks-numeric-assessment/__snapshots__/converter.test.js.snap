// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericAssessment Converter oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "data": Object {
    "content": Object {
      "numericChoices": Array [
        Object {
          "answer": "3",
          "feedback": Object {
            "children": Array [
              Object {
                "children": Array [],
                "content": Object {
                  "textGroup": Array [
                    Object {
                      "data": Object {
                        "indent": 0,
                      },
                      "text": Object {
                        "styleList": Array [],
                        "value": "3222",
                      },
                    },
                  ],
                },
                "id": "text_id",
                "type": "ObojoboDraft.Chunks.Text",
              },
            ],
            "content": Object {},
            "id": "feedback_id",
            "type": "ObojoboDraft.Chunks.NumericAssessment.NumericFeedback",
          },
          "margin": "3",
          "requirement": "margin",
          "score": 100,
          "type": "percent",
        },
        Object {
          "answer": "3",
          "margin": "3",
          "requirement": "margin",
          "score": 100,
          "type": "absolute",
        },
      ],
    },
  },
  "key": "mock_id",
  "nodes": Array [
    Object {
      "nodes": Array [
        Object {
          "data": Object {
            "numericChoice": Object {
              "answer": "3",
              "feedback": Object {
                "children": Array [
                  Object {
                    "children": Array [],
                    "content": Object {
                      "textGroup": Array [
                        Object {
                          "data": Object {
                            "indent": 0,
                          },
                          "text": Object {
                            "styleList": Array [],
                            "value": "3222",
                          },
                        },
                      ],
                    },
                    "id": "text_id",
                    "type": "ObojoboDraft.Chunks.Text",
                  },
                ],
                "content": Object {},
                "id": "feedback_id",
                "type": "ObojoboDraft.Chunks.NumericAssessment.NumericFeedback",
              },
              "margin": "3",
              "requirement": "margin",
              "score": 100,
              "type": "percent",
            },
          },
          "object": "block",
          "type": "ObojoboDraft.Chunks.NumericAssessment.NumericAsnwer",
        },
        undefined,
      ],
      "object": "block",
      "type": "ObojoboDraft.Chunks.NumericAssessment.NumericChoice",
    },
    Object {
      "nodes": Array [
        Object {
          "data": Object {
            "numericChoice": Object {
              "answer": "3",
              "margin": "3",
              "requirement": "margin",
              "score": 100,
              "type": "absolute",
            },
          },
          "object": "block",
          "type": "ObojoboDraft.Chunks.NumericAssessment.NumericAsnwer",
        },
      ],
      "object": "block",
      "type": "ObojoboDraft.Chunks.NumericAssessment.NumericChoice",
    },
  ],
  "object": "block",
  "type": "mock_type",
}
`;

exports[`NumericAssessment Converter oboToSlate converts an OboNode to a Slate node with no content 1`] = `
Object {
  "data": Object {
    "content": undefined,
  },
  "key": "mock_key",
  "nodes": Array [],
  "object": "block",
  "type": "mock_type",
}
`;

exports[`NumericAssessment Converter slateToObo converts a Slate node to an OboNode 1`] = `
Object {
  "children": Array [],
  "content": Object {
    "numericChoices": Array [
      Object {
        "feedback": undefined,
        "requirement": "exact",
      },
    ],
  },
  "id": "mock_key",
  "type": "mock_type",
}
`;

exports[`NumericAssessment Converter slateToObo converts a Slate node to an OboNode with no content 1`] = `
Object {
  "children": Array [],
  "content": Object {
    "numericChoices": Array [],
  },
  "id": "mock_key",
  "type": "mock_type",
}
`;

exports[`NumericAssessment Converter slateToObo converts a Slate node to an OboNode with two correct 1`] = `
Object {
  "children": Array [],
  "content": Object {
    "numericChoices": Array [],
  },
  "id": "mock_key",
  "type": "mock_type",
}
`;
